app.js

```diff
@@ -78,72 +69,50 @@ app.get("/*", (req, res) => {
 
   lookup(decodeURI(slug)).then(
     (destination) => {
-      const resultURL = destination
       var resultQuery = combineQueries(
-        querystring.parse(new URL(destination).searchParams.toString()),
+        querystring.parse(new URL(destination).search),
         query
       );
-      const parsedDestination = new URL(destination)
-      const url = parsedDestination.origin + parsedDestination.pathname + resultQuery + parsedDestination.hash
-      console.log({ url })
-      res.redirect(302, url)
+      const parsedDestination = new URL(destination);
+      const finalURL = parsedDestination.origin + parsedDestination.pathname + resultQuery + parsedDestination.hash
+      console.log({ finalURL })
+      res.redirect(302, finalURL)
```

Comment:
Small suggestion on this is to use 307 instead of 302 here so that the original method (POST/GET) is preserved. 

302: 
> For historical reasons, a user agent MAY change the request method from POST to GET for the subsequent request. If this behavior is undesired, the 307 (Temporary Redirect) status code can be used instead.